### DISCUSSION TITLE
Paintings spawning too late

### DISCUSSION THREAD

**Discussion by AlexanderMandera** [Minestom/Minestom#874] [Help]
I currently implemented a loader for the entity/*.mca files to extract paintings, so I can spawn them manually.
The painting are read correctly, I am spawning them like that inside the `InstanceChunkLoadEvent`:

```java
Entity e = new Entity(EntityType.PAINTING);
PaintingMeta meta = (PaintingMeta) e.getEntityMeta();

meta.setMotive(PaintingMeta.Motive.PIGSCENE);
meta.setDirection(Direction.WEST);

e.setInstance(instance, new Position(x, y, z)); // Using precise coordinates from MCA file
```

After joining, I see that they are spawned in code, but if I teleport to the location, the painting is not appearing.

At the time of writing, 1-2 minutes after, the painting appeared like normal.
Is there a solution, so the paintings are spawned directly?

Thank you very much in advance.

**Comment by erikzimmermann** [comment]
Maybe the packets of those paintings are buffered so they won't be fired instantly...

**Comment by AlexanderMandera** [comment]
I found out that the painting appears faster when logging in again, so it only took about 10-20s to appear.
Also the painting disappears again if I am leaving the view range (got ~500 blocks away) and walking back into the chunk again.

So maybe it maybe has something to do with the EntityTracker?

**Comment by iam4722202468** [comment]
Since you're setting `meta.setDirection(Direction.WEST)`, you might need to also set the entity rotation, see this issue https://github.com/Minestom/Minestom/issues/760. I don't think that's the only problem though, I'm also having issues with entity tracking not sending spawn packets.

**Reply by AlexanderMandera** [reply]
I have added the rotation too, but it also results in significant delay before spawning.
So I may need to look further into the EntityTracker to find the issue...

**Comment by AlexanderMandera** [comment]
I have found a synchronization cooldown of 1 minute:

https://github.com/Minestom/Minestom/blob/b6ba6901edbceda455755962287bd261e6707f41/src/main/java/net/minestom/server/entity/Entity.java#L544
It sends a EntityTeleportPacket every minute with the current position.

Additionally, the SpawnPaintingPacket doesn't send the right coordinate (it sends the painting coordinates and not the block location of the center). The painting appears because of this cooldown because it gets teleport to the given (right) position.

I will try to make a PR to fix this.
